jena-arq/src/main/java/org/apache/jena/sparql/algebra/optimize/TransformPathFlattern.java

```diff
@@ -53,9 +58,60 @@ public TransformPathFlattern(PathCompiler pathCompiler)
     @Override
     public Op transform(OpPath opPath)
     {
-        // Flatten down to triples where possible.
-        PathBlock pattern = pathCompiler.reduce(opPath.getTriplePath()) ;
-        // Any generated paths of exactly one to triple; convert to Op.
-        return PathLib.pathToTriples(pattern) ;
+        Op op = transformReduce(opPath, pathCompiler);
+        return op;
     }
-}
+
+    static Op transformReduce(Op op, PathCompiler pathCompiler) {
+        return Transformer.transform(
```

Comment:
I'm confused - this is only called with an `OpPath` as the first argument.

Reply:
so am I! fixed